### ISSUE TITLE
Errors reported twice by dlint.

### ISSUE THREAD

**Issue by lawrence-dol** [denoland/deno_lint#750]
### Lint Name

All, or at least most.

### Code Snippet

```js
for(var x;;);
```

### Expected Result

````
error[no-inner-declarations]: Move variable declaration to module root
 --> C:\...\test-script.js:1:5
  |
1 | for(var x;;);
  |     ^^^^^
  |
  = help: Move the declaration up into the correct scope
Found 1 problem
````

### Actual Result

````
error[no-inner-declarations]: Move variable declaration to module root
 --> C:\...\test-script.js:1:5
  |
1 | for(var x;;);
  |     ^^^^^
  |
  = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to module root
 --> C:\...\test-script.js:1:5
  |
1 | for(var x;;);
  |     ^^^^^
  |
  = help: Move the declaration up into the correct scope
Found 2 problems
````

### Additional Info

This happens for almost all errors reported. 

It's strange that this has not be noticed before. Perhaps this is unique to the Windows executable?

Just in case it helps to see more examples, I've added an additional, much larger, output from a module that is constrained to ES3 which produces a variety of linting errors (in part because it's been coded for easy transformation to ES2015+ as soon as we can get rid of a very old host-side JS runtime in production).

### Version

dlint 0.6.0

### Additional Results

````
C:\...\build\code>cls & dlint run --config=C:\...\build\$DenoLintRules.json C:\...\business\code\libweb\src\util\GeneralUtil.js

error[no-inner-declarations]: Move variable declaration to function root
  --> C:\...\src\util\GeneralUtil.js:90:9
   |
90 |     for(var xa=0; xa<fncs.length; ++xa) {
   |         ^^^^^^^^
   |
   = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
  --> C:\...\src\util\GeneralUtil.js:90:9
   |
90 |     for(var xa=0; xa<fncs.length; ++xa) {
   |         ^^^^^^^^
   |
   = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
  --> C:\...\src\util\GeneralUtil.js:99:17
   |
99 |             for(var xa=0,len=fncs.length; xa<len; ++xa) { fncs[xa].apply(null,args); }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
  --> C:\...\src\util\GeneralUtil.js:99:17
   |
99 |             for(var xa=0,len=fncs.length; xa<len; ++xa) { fncs[xa].apply(null,args); }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:189:17
    |
189 |             for(var xa=0,len=cmpfncs.length; xa<len && cmpres==0; ++xa) { cmpres=cmpfncs[xa](onercd,tworcd); }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:189:17
    |
189 |             for(var xa=0,len=cmpfncs.length; xa<len && cmpres==0; ++xa) { cmpres=cmpfncs[xa](onercd,tworcd); }
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-extra-semi]: Unnecessary semicolon.
   --> C:\...\src\util\GeneralUtil.js:230:10
    |
230 |         };
    |          ^
    |
    = help: Remove the extra (and unnecessary) semi-colon
error[no-extra-semi]: Unnecessary semicolon.
   --> C:\...\src\util\GeneralUtil.js:230:10
    |
230 |         };
    |          ^
    |
    = help: Remove the extra (and unnecessary) semi-colon
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:252:9
    |
252 |     for(var key in obj) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:252:9
    |
252 |     for(var key in obj) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
   --> C:\...\src\util\GeneralUtil.js:253:13
    |
253 |         if(!obj.hasOwnProperty(key)) { continue; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
   --> C:\...\src\util\GeneralUtil.js:253:13
    |
253 |         if(!obj.hasOwnProperty(key)) { continue; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:254:9
    |
254 |         var val=obj[key];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:254:9
    |
254 |         var val=obj[key];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:299:9
    |
299 |     for(var dft of argsArray(arguments,1)) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:299:9
    |
299 |     for(var dft of argsArray(arguments,1)) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:328:9
    |
328 |         var args=argsArray(arguments,2);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:328:9
    |
328 |         var args=argsArray(arguments,2);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:344:13
    |
344 |         for(var xa=0, len=arr.length; xa<len; ++xa) {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:344:13
    |
344 |         for(var xa=0, len=arr.length; xa<len; ++xa) {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:417:21
    |
417 |     if(exctyps) for(var xa=0,len=exctyps.length; xa<len; ++xa) {
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:417:21
    |
417 |     if(exctyps) for(var xa=0,len=exctyps.length; xa<len; ++xa) {
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:489:9
    |
489 |     for(var xa=1,len=arguments.length; xa<len; ++xa ) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:489:9
    |
489 |     for(var xa=1,len=arguments.length; xa<len; ++xa ) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:515:9
    |
515 |     for(var xa=1,len=arguments.length; xa<len; ++xa) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:515:9
    |
515 |     for(var xa=1,len=arguments.length; xa<len; ++xa) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:517:13
    |
517 |         for(var keys=Object.keys(src), xb=0; xb<keys.length; xb++) { var key=keys[xb]; tgt[key]=src[key]; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:517:70
    |
517 |         for(var keys=Object.keys(src), xb=0; xb<keys.length; xb++) { var key=keys[xb]; tgt[key]=src[key]; }
    |                                                                      ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:517:13
    |
517 |         for(var keys=Object.keys(src), xb=0; xb<keys.length; xb++) { var key=keys[xb]; tgt[key]=src[key]; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:517:70
    |
517 |         for(var keys=Object.keys(src), xb=0; xb<keys.length; xb++) { var key=keys[xb]; tgt[key]=src[key]; }
    |                                                                      ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:565:9
    |
565 |     for(var xa=1,len=arguments.length; xa<len; ++xa ) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:565:9
    |
565 |     for(var xa=1,len=arguments.length; xa<len; ++xa ) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-extra-semi]: Unnecessary semicolon.
   --> C:\...\src\util\GeneralUtil.js:634:10
    |
634 |         };
    |          ^
    |
    = help: Remove the extra (and unnecessary) semi-colon
error[no-extra-semi]: Unnecessary semicolon.
   --> C:\...\src\util\GeneralUtil.js:634:10
    |
634 |         };
    |          ^
    |
    = help: Remove the extra (and unnecessary) semi-colon
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:640:9
    |
640 |         var stgval=localStorage.getItem(stgnam);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:640:9
    |
640 |         var stgval=localStorage.getItem(stgnam);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-empty]: Empty block statement
   --> C:\...\src\util\GeneralUtil.js:642:75
    |
642 |             try { curval=JSON.parse(stgval).value || curval; } catch(err) {}                                            // ignore error, leaving current value unchanged
    |                                                                           ^^
    |
    = help: Add code or comment to the empty block
error[no-empty]: Empty block statement
   --> C:\...\src\util\GeneralUtil.js:642:75
    |
642 |             try { curval=JSON.parse(stgval).value || curval; } catch(err) {}                                            // ignore error, leaving current value unchanged
    |                                                                           ^^
    |
    = help: Add code or comment to the empty block
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:646:9
    |
646 |         var inival=curval;
    |         ^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:646:9
    |
646 |         var inival=curval;
    |         ^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:780:9
    |
780 |     for(var xa=0,len=pths.length; xa<len; ++xa) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:780:9
    |
780 |     for(var xa=0,len=pths.length; xa<len; ++xa) {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:781:9
    |
781 |         var pth=pths[xa];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:781:9
    |
781 |         var pth=pths[xa];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:782:9
    |
782 |         var spl=pth.lastIndexOf("/");
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:782:9
    |
782 |         var spl=pth.lastIndexOf("/");
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:783:9
    |
783 |         var par,lea;
    |         ^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:783:9
    |
783 |         var par,lea;
    |         ^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:839:9
    |
839 |         var args=argsArray(arguments,1);                                                                                // include `fnc` which will become the current count
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:839:9
    |
839 |         var args=argsArray(arguments,1);                                                                                // include `fnc` which will become the current count
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:842:13
    |
842 |             var lmt=cnt;
    |             ^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:842:13
    |
842 |             var lmt=cnt;
    |             ^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:845:13
    |
845 |         for(var xa=0; cnt(xa); ++xa) {
    |             ^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:845:13
    |
845 |         for(var xa=0; cnt(xa); ++xa) {
    |             ^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:986:9
    |
986 |     for(var key in obj) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:986:9
    |
986 |     for(var key in obj) {
    |         ^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
   --> C:\...\src\util\GeneralUtil.js:987:13
    |
987 |         if(!obj.hasOwnProperty(key)) { continue; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
   --> C:\...\src\util\GeneralUtil.js:987:13
    |
987 |         if(!obj.hasOwnProperty(key)) { continue; }
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:988:9
    |
988 |         var val=obj[key];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:988:9
    |
988 |         var val=obj[key];
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:998:13
    |
998 |         for(var xa=0,len=srcobj.length; xa<len; ++xa) {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:998:13
    |
998 |         for(var xa=0,len=srcobj.length; xa<len; ++xa) {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:999:13
    |
999 |             var srcval=srcobj[xa];
    |             ^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
   --> C:\...\src\util\GeneralUtil.js:999:13
    |
999 |             var srcval=srcobj[xa];
    |             ^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1016:13
     |
1016 |         for(var key in srcobj) {
     |             ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1016:13
     |
1016 |         for(var key in srcobj) {
     |             ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1017:17
     |
1017 |             if(!srcobj.hasOwnProperty(key)) { continue; }
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1017:17
     |
1017 |             if(!srcobj.hasOwnProperty(key)) { continue; }
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1018:13
     |
1018 |             var srcval=srcobj[key];
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-redeclare]: Redeclaration is not allowed
    --> C:\...\src\util\GeneralUtil.js:1018:17
     |
1018 |             var srcval=srcobj[key];
     |                 ^^^^^^
     |
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1018:13
     |
1018 |             var srcval=srcobj[key];
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-redeclare]: Redeclaration is not allowed
    --> C:\...\src\util\GeneralUtil.js:1018:17
     |
1018 |             var srcval=srcobj[key];
     |                 ^^^^^^
     |
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1040:9
     |
1040 |     for(var key in obj) {
     |         ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1040:9
     |
1040 |     for(var key in obj) {
     |         ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1041:13
     |
1041 |         if(!obj.hasOwnProperty(key)) { continue; }
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1041:13
     |
1041 |         if(!obj.hasOwnProperty(key)) { continue; }
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1042:9
     |
1042 |         var val=obj[key];
     |         ^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1042:9
     |
1042 |         var val=obj[key];
     |         ^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1054:9
     |
1054 |     for(var key in obj) {
     |         ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1054:9
     |
1054 |     for(var key in obj) {
     |         ^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1055:13
     |
1055 |         if(!obj.hasOwnProperty(key)) { continue; }
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-prototype-builtins]: Access to Object.prototype.hasOwnProperty is not allowed from target object
    --> C:\...\src\util\GeneralUtil.js:1055:13
     |
1055 |         if(!obj.hasOwnProperty(key)) { continue; }
     |             ^^^^^^^^^^^^^^^^^^^^^^^
     |
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1056:9
     |
1056 |         var val     = obj[key];
     |         ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
error[no-inner-declarations]: Move variable declaration to function root
    --> C:\...\src\util\GeneralUtil.js:1056:9
     |
1056 |         var val     = obj[key];
     |         ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: Move the declaration up into the correct scope
Found 84 problems

C:\...\build\code>
````

**Comment by magurotuna** [comment]
@lawrence-dol Thanks. Is this issue specific to dlint? That is, if you run `deno lint` rather than `dlint run`, it doesn't happen, right?
Unfortunately, I don't have a Windows machine so I think it will likely take a while to figure out the cause. Please give me a moment.

**Comment by magurotuna** [comment]
Hmm, I attempted to run `dlint` in the windows runner on GitHub Actions, but I didn't see the issue happen.
https://github.com/magurotuna/deno_lint/pull/1/checks?check_run_id=2895483816#step:8:294

So I suspect it's not related to Windows. Not quite sure, but it might be because of the config file. Can I have a look at the content of your `DenoLintRules.json`?

**Comment by lawrence-dol** [comment]
Confirmed that the problem is only with `dlint` and does not happen with `deno lint`.

DenoLintRules.json (BTW, what does `tags` do? I just copied from examples for this file.):

````
{
"rules": {
    "tags": ["recommended"],
    "include": [
        "adjacent-overload-signatures",
        "ban-ts-comment",
        "ban-types",
        "ban-untagged-ignore",
        "ban-untagged-todo",
        "constructor-super",
        "for-direction",
        "getter-return",
        "no-case-declarations",
        "no-class-assign",
        "no-compare-neg-zero",
        "no-constant-condition",
        "no-control-regex",
        "no-debugger",
        "no-delete-var",
        "no-deprecated-deno-api",
        "no-dupe-args",
        "no-dupe-class-members",
        "no-dupe-else-if",
        "no-dupe-keys",
        "no-duplicate-case",
        "no-empty-character-class",
        "no-empty-interface",
        "no-empty-pattern",
        "no-empty",
        "no-ex-assign",
        "no-explicit-any",
        "no-extra-boolean-cast",
        "no-extra-non-null-assertion",
        "no-extra-semi",
        "no-fallthrough",
        "no-func-assign",
        "no-global-assign",
        "no-import-assign",
        "no-inferrable-types",
        "no-inner-declarations",
        "no-invalid-regexp",
        "no-invalid-triple-slash-reference",
        "no-irregular-whitespace",
        "no-misused-new",
        "no-mixed-spaces-and-tabs",
        "no-namespace",
        "no-new-symbol",
        "no-obj-calls",
        "no-octal",
        "no-prototype-builtins",
        "no-redeclare",
        "no-regex-spaces",
        "no-self-assign",
        "no-setter-return",
        "no-shadow-restricted-names",
        "no-this-alias",
        "no-this-before-super",
        "no-unreachable",
        "no-unsafe-finally",
        "no-unsafe-negation",
        "no-with",
        "prefer-as-const",
        "prefer-namespace-keyword",
        "require-await",
        "require-yield",
        "use-isnan",
        "valid-typeof",
        ""
        ],
    "exclude": [
        "camelcase",
        "no-cond-assign",
        "no-unused-labels",
        "no-unused-vars",
        "prefer-const",
        ""
        ]
    }
}
````

**Comment by magurotuna** [comment]
@lawrence-dol 
Thanks for the additional info, I got it. The "recommended" tag enables the set of recommended rules, which you can see at [lint.deno.land](https://lint.deno.land/) with "Show only recommended rules" on. And you enable a bunch of rules in `include` section, some (or perhaps all?) of which are also tagged with "recommended and therefore it results in the same rule being run twice.
Maybe `dlint` should avoid running the same rule multiple times even if `tags` and `include` sections are conflict. I'll fix it.

**Comment by lawrence-dol** [comment]
@magurotuna : Doh! Thanks!

Yes, I do have all the rules listed, as either included or excluded, and that was my intent.

**Comment by lawrence-dol** [comment]
@magurotuna : A few questions, then.

Have the contents of the config file been documented yet, and can you point me to it? 

I've noted that the output of `dlint rules` and `deno lint --rules` appear to be incomplete (e.g. `no-vars` is missing). Will there be a reliable way to get a complete list of the currently supported rules eventually, or should I expect to add rules with new versions as I encounter them?

Is there a way to effectively turn off all linting rules? Sounds stupid, I know, but I have one use case, that of checking for syntactic correctness (no missing semicolons) in minimized files, which I just want to subject to syntax validation during build. So I just want to piggy back on the fact that `dlint` has to parse the script, without executing it, before linting. (Using `deno run` fails because it tries to resolve dependencies which are present until all scripts are loaded in the browser.)

**Comment by magurotuna** [comment]
@lawrence-dol 
There's no documentation about the config file but ideally it should. I'll add it to my TODO list 👍 

> I've noted that the output of dlint rules and deno lint --rules appear to be incomplete (e.g. no-var is missing). Will there be a reliable way to get a complete list of the currently supported rules eventually, or should I expect to add rules with new versions as I encounter them?

`deno lint` runs the "recommended" rules only, thus running `deno lint --rules` shows you the "recommended" only. On the other hand, `dlint rules` shows all the implemented rules.
In `deno lint`, the reliable way is `deno lint --rules`; In `dlint`, it is `dlint rules`. 
There are documentations about the rules, such as

- https://deno.land/manual@v1.11.2/tools/linter
- https://lint.deno.land/

but these are sometimes outdated or preceding although we take care to make it be compatible.

> Is there a way to effectively turn off all linting rules? 

In #753, I also changed how the config file works when nothing specified (i.e. `tags`, `include`, and `exclude` are all empty) so that no rules are run by default. So after the new version of `dlint` is released, you'll able to turn off all rules with the config like:

```json
{
  "rules": {
    "tags": [],
    "include": [],
    "exclude": []
  }
}
```

> Sounds stupid, I know, but I have one use case, that of checking for syntactic correctness (no missing semicolons) in minimized files, which I just want to subject to syntax validation during build. 

Why not using `deno fmt --check` for syntax validation? I think it should work pretty fine as well.

**Comment by lawrence-dol** [comment]
@magurotuna 👍 👍 👍 

>  `dlint rules` shows all the implemented rules.

That appears not to be the case -- I suspect it's showing only the recommended rules. Certainly, `no-var`, `eqeqeq`, `explicit-function-return-type`, and `explicit-module-boundary-types` are not being listed in 0.6.0.

**Comment by magurotuna** [comment]
@lawrence-dol 
Ooops. Actually `--all` is required to get all the rules, like `dlint rules --all`. Please try this out :)

And FYI, each dlint command can output help messages. For instance, running `dlint run --help` outputs:

```
dlint-run

USAGE:
    dlint run [OPTIONS] [--] [FILES]...

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
        --config <CONFIG>       Load config from file
        --plugin <PLUGIN>...    Specify plugin paths
        --rule <RULE_CODE>      Run a certain rule

ARGS:
    <FILES>...    Set the input file to use
```

or `dlint rules --help`:

```
dlint-rules 

USAGE:
    dlint rules [FLAGS] [RULE_NAME]

FLAGS:
        --all        
    -h, --help       Prints help information
        --json       
    -V, --version    Prints version information

ARGS:
    <RULE_NAME>    Show detailed information about rule
```

These help messages might be helpful to some extent.

**Comment by lawrence-dol** [comment]
> each dlint command can output help messages.

Nuts! I didn't think to try that. Thanks for all the help!

**Comment by lawrence-dol** [comment]
Hmmm. I don't suppose `"tags": [ "all" ], "exclude": ... ]` will enable all rules except those in the exclude list?

**Comment by magurotuna** [comment]
"all" is not a tag. Currently the only valid tag is "recommended".
And the application order of the config is `tag` -> `exclude` -> `include`. For example, if we use the following config, `eqeqeq` is enabled eventually.

```json
{
  "rules": {
    "tags": [],
    "include": ["eqeqeq"],
    "exclude": ["eqeqeq"]
  }
}
```

The point is, if you want to enable as many rules as possible, the config file would look like:

```jsonc
{
  "rules": {
    "tags": ["recommended"],
    "include": ["no-await-in-loop", ...], // specify the rules that are not in the "recommended" set and that you want to enable
    "exclude": ["no-explicit-any", ...] // specify the rules that you want to disable from "recommended"
  }
}
```

Note that the rules that don't belong to "recommended" are kind of prone to false positives/negatives, because they are poorly maintained and we have got few feedbacks from the users since they are disabled in `deno lint`. Please feel free to open issues if you run into any problems :)
